Automattic/sensei
# Enable Learning Mode blocks to be configured by theme.json

Depends on https://github.com/Automattic/sensei/pull/6066

### Changes proposed in this Pull Request
* Update some learning mode blocks to enable get configuration from block-based themes.


### Context 
As part of the new sensei theme initiative,  we need to have the blocks getting more style configurations  (typography, spacing... ) from theme.json files and configurable by the editor.


### Testing instructions
* Switch to the branch
* run `npm run start`
* Go to the `Lesson (Learning Mode - Default)` template editor
* Check if any block is broken
* Check is possible to change the font family, font weight, font style, and decoration on the editor.
*  Make a change, save and publish and check if the rendered version is following the same style applied on the editor:

Affected Blocks:

* Course Navigation
* Course-theme-course-progress-counter (information on the header)
* Course-theme-lesson-module (Module title above the lesson name)
* Course Title (Header)
* Exit Course Link (Header)
*  Progress Bar (Above bellow the header)  <--- This one change is related to the received background colors from theme.json: 


You can also make changes by adding the following instructions using the `theme.json` on your active block based theme:


```
                   "sensei-lms/course-title" : {
				"typography": {
					"textTransform": "uppercase",
					"fontFamily": "var(--wp--preset--font-family--league-gothic)",
					"fontSize": "var(--wp--preset--font-size--medium)"
				}
			},
			
			"sensei-lms/course-theme-course-progress-bar": {
				"color": {
					"background": "#ffffff"
				}
			},
			"sensei-lms/course-theme-course-progress-counter" : {
				"typography": {
					"fontFamily": "var(--wp--preset--font-family--system)",
					"fontSize": "var(--wp--preset--font-size--xx-small)",
					"letterSpacing": "0.02em"
				}
			},
			"sensei-lms/exit-course" : {
				"typography": {
					"fontFamily": "var(--wp--preset--font-family--system)",
					"fontSize": "var(--wp--preset--font-size--xx-small)",
					"textDecoration": "underline",
					"letterSpacing": "0.02em"
				}
			},
 

```
